### PR TITLE
To Do list add & remove

### PR DESCRIPTION
**Pull request by brhanuhailu**
### To Do list: add & remove 
# This is pull request I applied:
- [X] The CRUD (create, read, update, delete) in detail
      1. Implemented a function for adding a new task (add a new element to the array)
      2. Implemented a function for deleting a task (remove an element from the array)
      3. Implement a function for editing task descriptions
- [X]  Lists are saved in local storage
- [X]  Webpack used to bundle JavaScript
- [X] Linter error checked

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,39 +1,124 @@
+/* eslint no-alert: "error" */
+/* eslint-disable max-classes-per-file */
+/* eslint-disable no-use-before-define */
+
```
Comment (GzimAsani) [inline src/index.js]: - [ ] Please remove the disabling linters, remember they're there to help you write better code. Writing a good and understandable code is very important

File: src/index.js [hunk 2]
```diff
@@ -1,39 +1,124 @@
...
+      });
+
+      li[element.index].addEventListener('mouseleave', (e) => {
+        button[element.index].innerHTML = '<i class="fa-solid fa-ellipsis-vertical"></i>';
+        this.updatetask(e.target.id, e.target.innerText);
+      });
+    });
+  }
+
+  addtask = (description, completed, index) => {
```
Comment (GzimAsani) [inline src/index.js]: - [ ] As stated on the requirements, please add a new JS file for the new functionality 
image

File: src/index.js [hunk 3]
```diff
@@ -1,39 +1,124 @@
...
+    } else {
+      this.taskDtata.splice(key, 1);
+    }
+    this.taskDtata.forEach((element, index) => {
+      element.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('LOCALLISTDB', JSON.stringify(this.taskDtata));
+      this.display();
+    }
```
Comment (GzimAsani) [inline src/index.js]: - [ ] The button should appear always for deleting a task, currently it appears only when I click on a specific todo
image

Comment (GzimAsani) [reply src/index.js]: - [ ] The remove function is not working correctly, kindly check the video https://www.loom.com/share/86adb67040054e2c97aa93a32df3ee22

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @brhanuhailu              :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with adding and updating functionalities :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 4]
```diff
@@ -1,39 +1,166 @@
...
+    });
+  }
+
+  addtask = (description, completed, index) => {
+    const newtask = new Tasks(description, completed, index);
+    this.taskDtata.push(newtask);
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('LOCALLISTDB', JSON.stringify(this.taskDtata));
+    }
+  }
```
Comment (ABDULALI3468) [inline src/index.js]: As requested by the previous reviewer it's a project requirement to have this functionality of **Adding tasks** in a new file. 

## Solution:--

Then you can import that function in this file and use it as you want.

But obviously, for all variables declared here can't use them in that file so you can use the function parameters to pass them.

File: src/index.js [hunk 5]
```diff
@@ -1,39 +1,166 @@
...
+      this.taskDtata.splice(key, 1);
+    }
+    this.taskDtata.forEach((element, index) => {
+      element.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('LOCALLISTDB', JSON.stringify(this.taskDtata));
+      this.display();
+    }
+  }
```
Comment (ABDULALI3468) [inline src/index.js]: As requested by the previous reviewer it's a project requirement to have this functionality of **Removing tasks** in a new file. 

## Solution:--

Then you can import that function in this file and use it as you want.

But obviously, for all variables declared here can't use them in that file so you can use the function parameters to pass them.

File: src/index.js [hunk 6]
```diff
@@ -1,39 +1,166 @@
...
+      this.taskDtata[item].description = description;
+    }
+    this.taskDtata.forEach((element, index) => {
+      element.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('LOCALLISTDB', JSON.stringify(this.taskDtata));
+      this.display();
+    }
+  }
```
Comment (ABDULALI3468) [inline src/index.js]: As requested by the previous reviewer it's a project requirement to have this functionality of **Updating tasks** in a new file. 

## Solution:--

Then you can import that function in this file and use it as you want.

But obviously, for all variables declared here can't use them in that file so you can use the function parameters to pass them.

**Review by ABDULALI3468 — CHANGES_REQUESTED** [review]
# Changes Required 👍 

Hi @brhanuhailu , 



Great work on making the changes requested by a previous reviewer 👏 
You've done really well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights 
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.


 ## Required Changes ♻️

> **_Check the comments below to have a look at the required changes._**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (wayungi) [reply src/index.js]: - [ ] Greate work re-enabling the linters 👏🏾

**Review by wayungi — APPROVED** [review]
Hi @brhanuhailu , 
image

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] Profesional README ✔️ 
- [ ] All linter checks passing ✔️ 
- [ ] Good commit messages ✔
- [ ] No merge conflicts ✔️ 
- [ ]  Correct GitHub flow ✔️ 
- [ ] Descriptive PR body 

## Optional suggestions
- [ OPTIONAL]  the property index should be set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5). This requirement has not been met. 
reqs
Your fourth element in the picture below 👇🏾 should have an index of four not three. It would be nice if you applied this change
index


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**